docs/source/developers/crossbow.rst

```diff
@@ -0,0 +1,284 @@
+.. raw:: html
+
+   <!--
+   Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   -->
+
+Packaging and Testing with Crossbow
+===================================
+
+The content of this directory aims for automating the historically error
+prone process of Arrow packaging and integration testing.
+
+Packages:
+  - C++ and Python `conda-forge packages`_ for Linux, Mac and Windows
+  - Python `Wheels`_ for Linux, Mac and Windows
+  - C++ and GLib `Linux packages`_ for multiple distributions
+  - Java for Gandiva
+
+Integration tests:
+  - Various docker tests
+  - Pandas
+  - Dask
+  - Turbodbc
+  - HDFS
+  - Spark
+
+Architecture
+------------
+
+Executors
+~~~~~~~~~
+
+Individual jobs are executed on public CI services, currently: - Linux:
+TravisCI - Mac: TravisCI - Windows: AppVeyor
```

Comment:
It appears that we're actually using Circle CI? https://github.com/ursa-labs/crossbow/branches/all?page=2&query=nightly-625&utf8=%E2%9C%93